code/game/objects/effects/temporary_visuals/miscellaneous.dm

```diff
@@ -6,16 +6,22 @@
 /obj/effect/temp_visual/explosion
 	name = "explosion"
 	icon = 'icons/effects/explosion.dmi'
-	icon_state = "explosion"
+	icon_state = ""
 	duration = 8
-	pixel_x = -16
-	pixel_y = -16
 	light_system = STATIC_LIGHT
+	///The icon for the explosion itself
+	var/explosion_icon = "explosion"
```

Comment:
couldnt you just set icon_state then null this in initialize

Reply:
done